components/script/dom/domimplementation.rs

```diff
@@ -75,11 +75,22 @@ impl DOMImplementationMethods for DOMImplementation {
             _ => "application/xml"
         };
 
+        // @TODO What's the right thing to do for this? Without this, tests in DOMImplementation-createDocument.html
+        // and Node-properties.html fail.
+        //
+        // Before, the call to XMLDocument::new was hardcoded to pass in NonHTMLDocument as the doc type. That means
+        // that document.is_html_document is false for documents that are actually HTML. That makes the tests fail
+        // because I use that property in document.createElement to decide whether to use the HTML namespace or not.
+        let doc_type = match maybe_doctype {
```

Comment:
This does look problematic; for example, consider the case
```js
var doctype = document.implementation.createDocumentType("html", "", "");
var doc = document.implementation.createDocument("http://www.w3.org/1999/xhtml", "html", doctype);
assert_equals(doc.documentElement.tagName, "html")
```

I believe this should pass per spec, and it does in Chrome, though not in Gecko. We should add this test somewhere.